File: ios/Talise/Features/Send/SendCompleteView.swift
import SwiftUI
import UIKit

/// Step 5: success. Renders the Figma "Successful PopUp" (node 132:2)
/// celebration — pastel-green field, coin-stack illustration, the sent
/// amount in the user's currency, "gas cost = 0 / money arrives < 1s",
/// and a Share Receipt + Done row. `onDone` lets the parent dismiss the
/// whole Send flow.
struct SendCompleteView: View {
    @Bindable var draft: SendDraft
    var onDone: () -> Void

    var body: some View {
        // Amount in the user's display currency (USDsui is 1:1 USD).
        // local2 mirrors what the rest of the app shows so a ₦ user
        // sees ₦ here, a $ user sees $.
        let amountText = TaliseFormat.local2(draft.success?.usdsui ?? draft.amountUsdsui)
        SuccessfulTxView(
            amountText: amountText,
            onShareReceipt: shareReceipt,
            onDone: onDone
        )
        .toolbar(.hidden, for: .navigationBar)
    }

    /// Share the on-chain explorer link for this payment via the system
    /// share sheet. No-op if we somehow reached this screen without a
    /// digest (shouldn't happen — .complete is gated on a real digest).
    private func shareReceipt() {
        guard let digest = draft.success?.digest, !digest.isEmpty else { return }
        let url = "https://suivision.xyz/txblock/\(digest)"
        let av = UIActivityViewController(
            activityItems: [URL(string: url) ?? url],
            applicationActivities: nil
        )
        guard let scene = UIApplication.shared.connectedScenes
                .compactMap({ $0 as? UIWindowScene })
                .first(where: { $0.activationState == .foregroundActive }),
              let root = scene.keyWindow?.rootViewController else { return }
        // Walk to the top-most presented controller so the share sheet
        // mounts above the Send fullScreenCover.
        var top = root
        while let presented = top.presentedViewController { top = presented }
        av.popoverPresentationController?.sourceView = top.view
        top.present(av, animated: true)
    }
}

File: android/app/src/main/java/io/talise/app/feature/send/SendCompleteView.kt
package io.talise.app.feature.send

import android.content.Intent
import androidx.compose.runtime.Composable
import androidx.compose.ui.platform.LocalContext

/**
 * Step 5: success — iOS `SendCompleteView`. Renders the Figma "Successful
 * PopUp" celebration via [SuccessfulTxView]: the sent amount in the user's
 * currency, "gas cost = 0, money arrives < 1s", and Share Receipt + Done.
 */
@Composable
fun SendCompleteView(
    draft: SendDraft,
    onDone: () -> Unit,
) {
    val context = LocalContext.current

    // Amount in the user's display currency (USDsui is 1:1 USD).
    val amountText = sendLocal2(draft.success?.usdsui ?: draft.amountUsdsui)
    // Spend + Save pop: the server-blessed round-up amount that auto-saved
    // with this send (0 → no pop).
    val savedUsd = draft.success?.savedUsd ?: 0.0
    val crossCurrency = draft.isCrossCurrency

    // Headline: same-currency wallet sends keep the default celebration;
    // cross-border fiat-payout sends say "Sent" to mark the chain leg as
    // final WITHOUT overclaiming bank delivery.
    val title = if (crossCurrency) "Sent" else "Transaction Successful!"
    val subtitle = if (crossCurrency) {
        val name = draft.success?.recipientDisplay ?: "their bank"
        "Sent, on its way to $name's bank"
    } else {
        "gas cost = 0, money arrives < 1s"
    }

    SuccessfulTxView(
        amountText = amountText,
        title = title,
        subtitle = subtitle,
        onShareReceipt = {
            // Share the on-chain explorer link via the system share sheet.
            // No-op without a digest (shouldn't happen — Complete is gated
            // on a real digest).
            val digest = draft.success?.digest
            if (!digest.isNullOrEmpty()) {
                val intent = Intent(Intent.ACTION_SEND).apply {
                    type = "text/plain"
                    putExtra(Intent.EXTRA_TEXT, "https://suivision.xyz/txblock/$digest")
                }
                context.startActivity(Intent.createChooser(intent, null))
            }
        },
        onDone = onDone,
        savedText = if (savedUsd > 0) sendLocal2(savedUsd) else null,
        // Only same-currency wallet sends are chain-final end-to-end, so
        // only they carry the recipient display; cross-border passes null.
        recipientDisplay = if (crossCurrency) null else draft.success?.recipientDisplay,
    )
}
